src/transactionIntentsFactories/delegationTransactionIntentsFactory.spec.ts

```diff
@@ -0,0 +1,301 @@
+import BigNumber from "bignumber.js";
+import { Address } from "../address";
+import { DelegationTransactionIntentsFactory } from "./delegationTransactionIntentsFactory";
+import { assert } from "chai";
+import { DELEGATION_MANAGER_SC_ADDRESS } from "../constants";
+import { ValidatorPublicKey } from "@multiversx/sdk-wallet-next";
+import { SignableMessage } from "../signableMessage";
+
+describe("test delegation intents factory", function () {
+    const delegationFactory = new DelegationTransactionIntentsFactory({
+        chainID: "D",
+        minGasLimit: 50000,
+        gasLimitPerByte: 1500,
+        gasLimitStake: 5000000,
+        gasLimitUnstake: 5000000,
+        gasLimitUnbond: 5000000,
+        gasLimitCreateDelegationContract: 50000000,
+        gasLimitDelegationOperations: 1000000,
+        additionalGasLimitPerValidatorNode: 6000000,
+        additionalGasLimitForDelegationOperations: 10000000
+    });
+
+    it("should build intent for new delegation contract", async function () {
```

Comment:
Now I see, we can change "should build" to "should create" (optional, "philosophical").

Reply:
Changed to `should create`